### PR TITLE
Pre release 1.0

### PR DESCRIPTION
**Pull request by benjaminaubin**
- Fixed package versions
- Removed taming-transformer deps by adding the corresponding missing files for LPIPS loss
- Cleaned with isort and black #52

### CODE REVIEW

File: sgm/modules/autoencoding/losses/__init__.py [hunk 1]
```diff
@@ -3,10 +3,9 @@
 import torch
 import torch.nn as nn
 from einops import rearrange
-from taming.modules.discriminator.model import NLayerDiscriminator, weights_init
-from taming.modules.losses.lpips import LPIPS
-from taming.modules.losses.vqperceptual import hinge_d_loss, vanilla_d_loss
-
+from ...diffusionmodules.lpips.lpips import LPIPS
```
Comment (ablattmann) [inline sgm/modules/autoencoding/losses/__init__.py:8]: I would rather place the `lpips` directory in `sgm/modules/autoending` since autoencoder training is where nearly all the classes in LPIPS are actually used

**Comment by akx** [comment]
The LPIPS vendoring isn't necessary if #50 is merged.

File: sgm/__init__.py [hunk 2]
```diff
@@ -1,5 +1,4 @@
-from .data import StableDataModuleFromConfig
```
Comment (akx) [inline sgm/__init__.py:1]: So this was grabbed from #44? ;-)

File: sgm/modules/diffusionmodules/lpips/lpips.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
+"""Stripped version of https://github.com/richzhang/PerceptualSimilarity/tree/master/models"""
```
Comment (akx) [inline sgm/modules/diffusionmodules/lpips/lpips.py:1]: https://github.com/richzhang/PerceptualSimilarity/blob/master/LICENSE

File: requirements/pt13.txt
Comment (ablattmann) [inline requirements/pt13.txt:1]: Did you test building the environment with those package versions and starting the trainings in `configs/example_training/`?

File: sgm/modules/diffusionmodules/lpips/lpips.py (hunk 3 above)
Comment (benjaminaubin) [reply sgm/modules/diffusionmodules/lpips/lpips.py:1]: Good point should add it

File: sgm/modules/diffusionmodules/lpips/model.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    if classname.find("Conv") != -1:
+        nn.init.normal_(m.weight.data, 0.0, 0.02)
+    elif classname.find("BatchNorm") != -1:
+        nn.init.normal_(m.weight.data, 1.0, 0.02)
+        nn.init.constant_(m.bias.data, 0)
+
+
+class NLayerDiscriminator(nn.Module):
+    """Defines a PatchGAN discriminator as in Pix2Pix
+    --> see https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/models/networks.py
```
Comment (akx) [inline sgm/modules/diffusionmodules/lpips/model.py:19]: https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/LICENSE

File: requirements/pt2.txt
Comment (ablattmann) [inline requirements/pt2.txt:1]: same question as above

File: requirements/pt13.txt
Comment (ablattmann) [reply requirements/pt13.txt:1]: Especially important would be to run an autoencoder training and see if it still works

**Review by ablattmann — CHANGES_REQUESTED** [review]

**Comment by benjaminaubin** [comment]
> The LPIPS vendoring isn't necessary if #50 is merged.
Yes but we decided internally to not merge #50 for the future release
This LPIPS refactoring would have been necessary at some point in any case

File: sgm/modules/autoencoding/losses/__init__.py (hunk 1 above)
Comment (benjaminaubin) [reply sgm/modules/autoencoding/losses/__init__.py:8]: Can do that

File: sgm/modules/diffusionmodules/lpips/model.py (hunk 4 above)
Comment (benjaminaubin) [reply sgm/modules/diffusionmodules/lpips/model.py:19]: Done

File: sgm/modules/autoencoding/losses/__init__.py (hunk 1 above)
Comment (benjaminaubin) [reply sgm/modules/autoencoding/losses/__init__.py]: Done

File: requirements/pt2.txt
Comment (benjaminaubin) [reply requirements/pt2.txt:1]: What could be the issues? I simply install without fixing the versions, then fix the versions found by pip

File: sgm/__init__.py (hunk 2 above)
Comment (akx) [reply sgm/__init__.py:1]: Right. So could #44 please be merged, so there's only one way to import things from `sgm`?

File: sgm/modules/autoencoding/lpips/vqperceptual.py [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+def vanilla_d_loss(logits_real, logits_fake):
+    d_loss = 0.5 * (
+        torch.mean(torch.nn.functional.softplus(-logits_real))
+        + torch.mean(torch.nn.functional.softplus(logits_fake))
+    )
+    return d_loss
+
+
+class VQLPIPSWithDiscriminator(nn.Module):
```
[the remark this answers is not in the document]
Comment (benjaminaubin) [reply sgm/modules/autoencoding/lpips/vqperceptual.py:35]: yes indeed I shall clean this file

File: requirements/pt13.txt
Comment (palp) [reply requirements/pt13.txt:1]: I would love to build training tests that run through pytest to run on the cluster as part of #56 or a follow-up

Comment (benjaminaubin) [reply requirements/pt13.txt:1]: Please do @palp !

**Review by palp — APPROVED** [review]
I've merged this with #57 on my fork [here](https://github.com/palp/generative-models/tree/palp/1-0-pre-merge) and confirmed the tests pass & upstream dependencies are working, so this looks good from my perspective.

**Review by ablattmann — APPROVED** [review]

**Review by palp — APPROVED** [review]
Looks good, passes tests on #57
